ffi/tests/ctests/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+#[no_mangle]
+unsafe extern "C" fn rust_assert(
+    check: bool,
+    file: *const std::os::raw::c_char,
+    line: usize,
+    msg: *const std::os::raw::c_char,
+) {
+    assert!(
+        check,
+        "[{}:{}] {}",
+        std::ffi::CStr::from_ptr(file).to_str().unwrap(),
```

Comment:
It's a good idea to show filenames, but I don't think you need to pass them manually - unless I'm mistaken, with this approach if you run tests with `RUST_BACKTRACE=1 cargo test` you should already see C function, filename and line number that caused it to fail.

Reply:
You are right, i read the backtrace too quickly